src/v/archival/ntp_archiver_service.cc

```diff
@@ -948,6 +954,49 @@ ss::future<cloud_storage::upload_result> ntp_archiver::upload_segment(
       fib,
       lazy_abort_source,
       _segment_tags);
+}
+
+ss::future<cloud_storage::upload_result>
+ntp_archiver::stream_reference::close_on_cleanup_with_res(
```

Comment:
nit: I find it a bit odd that we pass in an `upload_result` only to return it. It could be returned inline after the call to `close_cleanup_with_res`.

Reply:
Yeah, this method was changed a couple of times and it ended up here (it earlier logged the result mostly for debugging a failing test). I will remove the result from this now, it is not required here.